Doc/library/argparse.rst

```diff
@@ -683,6 +744,8 @@ The add_argument() method
 The following sections describe how each of these are used.
 
 
+.. _name or flags:
+
 name or flags
 ^^^^^^^^^^^^^
```

Comment:
If I remember right, you don’t need to had «short link targets» to all sections here if you change the source of the links from `name_or_flags_` to
```
ref:`name or flags`
```

(I forget if it should be spaces or underscores)

Reply:
I went ahead and changed the line .. _name or flags: to .. ref: `name or flags`. However, I am unsure of the difference between the two. If I use the one that you suggested, what other things should I change in the file? 

Thank you!

Reply:
Sorry I did not explain the reStructuredText terminology enough.

What you added is manual references.  In order to create a link with a forward reference such as `prog_`, the target section needs to be marked with a declaration like `.. _prog`.

What I said is that every section title is automatically a reference target if you change the source of the links from `prog_` to
```
:ref:`prog`
```

Reply:
Thank you for the explanation! 
Should I go ahead and change all the lines that have this format (.. _prog) to this format (:ref:`prog`)?

Reply:
Well this is not to replace `.. _prog:`, but to replace `prog_`.

The `.. _prog:` part can be deleted!

The only negative aspect is that if the heading changes, the `ref` needs to be updated, which should not be a problem here (because the headings are just the argument names).

Reply:
First Option:
I removed the declarations that preceded the target sections (.. _prog). However, I found that if I changed the source of the links from prog_ to :ref:`prog`, I would get the following error message:
undefined label: (insert name here) (if the link has no caption the label must precede a section header). 
I would not get this error message if I removed the declarations (.. _prog) and used prog_ as the source of the links. 

Second Option: 
Based on this article (http://www.sphinx-doc.org/en/1.3/markup/inline.html#role-ref), I tried keeping the declaration (.. _prog) and changing all link sources to the :ref:`prog` format. This worked as well. 

I am unsure if I am on the wrong track - please feel free to correct me if I am! If I am on the right track, I wanted to ask if you had a preference for any of the two options above?

Reply:
Ah I see!

> undefined label: (insert name here) (if the link has no caption

A link with caption looks like this:

```
See the option :ref:`prog <prog>`
```

where the first `prog` is the link text (caption) and the second is the actual reference identifier.


At this point, it’s not worth your time to fight too much with reStructuredText, so do what feels the simplest and nicest option to you.

Reply:
I actually played around with using ``` :ref:`prog <prog>` ``` but received the same error message. Not sure if I am missing something? 

Sounds good! I am pretty new to reStructured Text so I will take your advice and the article's advice. I'll go ahead and change the links to the format you mentioned but keep the .. _prog in there. 

Thank you so much for your feedback and time on this! I really appreciate it!

Reply:
Ah well my goal was to make things simpler, if the first version using `prog_` and `.. _prog:` was the simplest one then you tried other things in vain!  Or maybe it will be useful to use `ref` since the custom caption makes it possible to have a flowing sentence in some case compared to `ref_`.

Reply:
I can also use ``` prog_ ``` and remove the ``` .. _prog: ``` if you prefer.

I was taking a look at the other text in the file and was wondering if I should avoid using ``` :ref: ``` to remain consistent with the rest of the text? Or, should I go ahead and use ``` ref ```?

Reply:
Having one way to do all references in the same file would be best.

I don’t have a preference; as you have tried a few variants, please feel free to decide which one is the nicest / cleanest to write and use only that one.

Reply:
I see. Thank you for the advice! 
Since most of the file uses the ``` prog_ ``` type reference, would it be okay to stick to that? I am afraid of potentially ruining any possible dependencies the original creator may have intended.

Reply:
Yes!

Reply:
Thanks merwok!

I've learned a lot more about reStructuredText from this so I am really thankful for your input and advice (: